extensions/amp-story/1.0/amp-story.js

```diff
@@ -1487,19 +1498,58 @@ export class AmpStory extends AMP.BaseElement {
       if (this.sidebar_ && sidebarState) {
         this.sidebarObserver_.observe(this.sidebar_, {attributes: true});
         actions.execute(this.sidebar_, 'open', /* args */ null,
-        /* source */ null, /* caller */ null, /* event */ null,
+            /* source */ null, /* caller */ null, /* event */ null,
             ActionTrust.HIGH);
+        this.openOpacityMask_();
       } else {
         this.sidebarObserver_.disconnect();
       }
     } else if (this.sidebar_ && sidebarState) {
       actions.execute(this.sidebar_, 'open', /* args */ null,
           /* source */ null, /* caller */ null, /* event */ null,
           ActionTrust.HIGH);
+      this.openOpacityMask_();
       this.storeService_.dispatch(Action.TOGGLE_SIDEBAR, false);
     }
   }
 
+  /**
+   * @private
+   */
+  openOpacityMask_() {
+    if (!this.maskElement_) {
+      const maskEl = this.win.document.createElement('div');
+      maskEl.classList.add(OPACITY_MASK_CLASS_NAME);
+      maskEl.addEventListener('click', () => {
+        const actions = Services.actionServiceForDoc(this.element);
+        if (this.sidebar_) {
```

Comment:
Was this `if` to make Closure Compiler happy? It looks like `openOpacityMask_` and `closeOpacityMask_` is only ever called from a function that already check `if (this.sidebar_)` so the check is redundant:

> https://github.com/ampproject/amphtml/blob/27bf8ae175dbf55a0fadc59a55a7432bb3fff3a8/extensions/amp-story/1.0/amp-story.js#L1498-L1503
`openOpacityMask_` and its calling function both check `if (this.sidebar_)` and only one should.

You can avoid redundant checks a couple of ways.

1. Add a non-nullable `!Element` parameter to the method and pass the reference when you call `openOpacityMask_` is called.
2. Use `dev().assertElement(this.sidebar_)` when you just need to make Closure Compiler happy. These calls are stripped at compile-time to avoid putting them in the final binary sent to the client.

This will make it super clear in the code when `this.sidebar_` is and isn't supposed to be defined.

Reply:
Thank you for catching this redundancy! The check was added in an earlier iteration, and I actually think that we should just be able to get rid of it entirely. I'll take notes on the different ways I can avoid redundancy for future PRs.